### ISSUE TITLE
Bizarre Softmax Behavior

### ISSUE THREAD

**Issue by gregoryquick** [gregoryquick/rust_perceptron#6]
A single softmax layer with squared error function on fixed batch does not produce any form consistent optimisation under GD, sometimes error increases sometimes it decreases, with large range of wandering.
A single softmax layer with cross entropy error function on fixed batch consistently goes to NaNs under GD.

**Comment by gregoryquick** [comment]
Should be fixed after changes in https://github.com/gregoryquick/rust_perceptron/commit/cfb9cdabe9722223df7d41974d96a15e75cf2573 and https://github.com/gregoryquick/rust_perceptron/commit/97911ab44e32ddb7987ac66cf42e7bb993e1aac6
